Restructure the documentation (#356)
pulp/pulp-cli: docs/installation.md

# Installation

The pulp-cli package can be installed from a variety of sources. After installing, see the next
section on how to configure pulp-cli.

## From PyPI

> 
> ```
> pip install pulp-cli  # minimal dependencies
> pip install pulp-cli[pygments]  # colorized output
> pip install pulp-cli[shell]  # with interactive shell mode
> ```

Reviewer: The way it is written, it looks like three commands that the user has to do rather than three options. 

```suggestion

To install with minimal dependencies:

pip install pulp-cli 

To install with colorized output:

pip install pulp-cli[pygments]

To install with an interactive shell:

pip install pulp-cli[shell]  # with interactive shell mode

For more information about the interactive shell, see ...

```
I had to remove the code blocks for the commands for my example to render.

Author: Actually it's four options. And they don't hurt if run one after another...
But yes we should split them.

fourth option: `pip install pulp-cli[pygments,shell]`

# Installation

The pulp-cli package can be installed from a variety of sources. After installing, see the next
section on how to configure pulp-cli.

## From PyPI

To install with minimal dependencies:
```bash
pip install pulp-cli  # minimal dependencies
```

You can also include optional dependencies for [advanced features](../advanced_features/) with one of the following commands:
```bash
pip install pulp-cli[pygments]  # colorized output
pip install pulp-cli[shell]  # with interactive shell mode
```

## From a source checkout

> If you intend to use unreleased features, or want to contribute to the CLI, you can perform a source installation:

Reviewer: ```suggestion
If you intend to use unreleased features, or want to contribute to the CLI, you can install from source:
```
